LabExT/Tests/Wafer/Export_data_test.py

```diff
@@ -0,0 +1,126 @@
+            "ExportJSON": ExportJSON,
+            "ExportCSV": ExportCSV,
+            "ExportHDF5": ExportHDF5,
+        }
+
+        file_name = join(dirname(dirname(__file__)), "example_measurement.json")
+        self.expm.exp.measurements = ObservableList()
+
+        with open(file_name) as f:
+            meas_dict = json.load(f)
+
+            # mock the loading of the measurement
+            # this is normally done in the ExperimentManager
+            meas_dict["file_path_known"] = file_name
+
+            for k in ["timestamp start", "timestamp", "timestamp end"]:
+                if k in meas_dict:
+                    meas_dict["timestamp_known"] = meas_dict[k]
+                    break
+          
+            for k in ["measurement name", "name"]:
+                if k in meas_dict:
+                    meas_dict["name_known"] = meas_dict[k]
+                    break
+           
+            meas_dict["measurement id long"] = uuid.uuid4().hex
+
+            for k in ["timestamp iso start", "timestamp_known"]:
+                if k in meas_dict:
+                    meas_dict["timestamp_iso_known"] = meas_dict[k]
+                    break
+
+            self.expm.exp.measurements.append(meas_dict)
+
+    def setup_window(self):
+        return ExportWizard(master=self.root, experiment_manager=self.expm)
+
+    @pytest.mark.flaky(reruns=3)
+    def test_create_window_instantiate_steps(self):
+        window = self.setup_window()
+        self.assertIsInstance(window.source_config_steps_insts["JavaScript Object Notation (.json)"], ExportJSON)
+        self.assertIsInstance(window.source_config_steps_insts["Comma-Separated Values (.csv)"], ExportCSV)
+        self.assertIsInstance(window.source_config_steps_insts["Hierarchical Data Format (.hdf5)"], ExportHDF5)
+
+    @pytest.mark.flaky(reruns=0)
+    def test_csv_format(self):
+        window = self.setup_window()
+        
+        window.current_step._select_all_button.invoke()
+        window.current_step.source_options_sel_var.set("Comma-Separated Values (.csv)")
+        self.pump_events()
+        
+        window._next_button.invoke()
+        self.pump_events()
+
+        def do_export():
+            window.current_step._export(window.selected_data)
+        
+        # normally, the export is done in a separate thread but this was causing issues with the test
+        window.current_step.on_next = do_export
+        
+        window.current_step.export_path = MockTKVar(dirname(dirname(__file__)))
+        window._next_button.invoke()
+        self.pump_events()
+
+        # file gets exported in a separate thread
+        window._next_button.invoke()
+        self.pump_events()
+
+        file_path = join(dirname(dirname(__file__)), "example_measurement.csv")
+
+        self.assertTrue(exists(file_path))
+
+        with open(file_path) as f:
+            data = f.read()
+            self.assertTrue(data.startswith("# CSV exported measurement data from LabExT\n"))
```

Comment:
Please add some more tests here, e.g. double check data content, other metadata?

Reply:
The test now checks the content of the data in the file against the measurement